python/dlr/api.py

```diff
@@ -342,3 +343,41 @@ def get_input(self, name, shape=None):
                                      out.ctypes.data_as(ctypes.POINTER(ctypes.c_float))))
         out = out.reshape(shape)
         return out
+
+# Interface
+class IDLRModel:
+    __metaclass__=abc.ABCMeta
+
+    @abc.abstractmethod
+    def get_input_names(self):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_input(self, name, shape=None):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def run(self, input_data):
+        raise NotImplementedError
+
+# Wrapper class
+class DLRModel(IDLRModel):
+    def __init__(self, model_path, dev_type='cpu', dev_id=0):
+        #Determine if 3rdparty package needed
+        for (dirpath, dirnames, filenames) in os.walk(model_path):
```

Comment:
This approach works only for folders. In case of TF model `model_path` will be (can be) .pb file (not folder)